superset/common/query_context.py

```diff
@@ -60,8 +60,8 @@ class QueryContext:
     queries: List[QueryObject]
     force: bool
     custom_cache_timeout: Optional[int]
-    result_type: utils.ChartDataResultType
-    result_format: utils.ChartDataResultFormat
+    result_type: Union[utils.ChartDataResultType, str]
+    result_format: Union[utils.ChartDataResultFormat, str]
```

Comment:
This `Union` shouldn't be necessary here. Are you getting linting errors without it?

Reply:
Yes, marshmallow schema validation was failing with the default (unspecified) values for `result_type` and `result_format`. Essentially, marshmallow and mypy cannot agree here. I tried using https://pypi.org/project/marshmallow-enum/ here as well, to no avail.

Reply:
I was able to get `marshmallow_enum` working, so the `Union` is no longer necessary.